lib/libspl/include/assert.h

```diff
@@ -56,10 +56,12 @@ do {									\
 	const TYPE __left = (TYPE)(LEFT);				\
 	const TYPE __right = (TYPE)(RIGHT);				\
 	if (!(__left OP __right)) {					\
-		char *__buf = alloca(256);				\
-		(void) snprintf(__buf, 256,				\
+		char *__buf = NULL;					\
+		int rc = asprintf(&__buf,				\
```

Comment:
asprintf will require a free. But I don't see why we need to do this. Why not pass the format strings and variable all the way, or pass va_list?